web/src/pages/profile.tsx

```diff
@@ -186,30 +186,31 @@ const ProfilePage = (props: Props): ReactElement => {
 
       setIsLoading(true);
 
-      sendOnSaveAnalytics(updateQueue.current().profileData, profileData);
+      sendOnSaveAnalytics(userData.profileData, profileData);
 
       if (profileData.employerId && profileData.employerId.length > 0) {
         updateQueue.queueTaskProgress({ [einTaskId]: "COMPLETED" });
       }
 
-      if (updateQueue.current().profileData.taxId !== profileData.taxId) {
+      if (userData.profileData.taxId !== profileData.taxId) {
         updateQueue.queueTaxFilingData({ state: undefined, registeredISO: undefined, filings: [] });
       }
 
-      if (updateQueue.current().profileData.industryId !== profileData.industryId) {
-        updateQueue.queueTaskProgress({ [naicsCodeTaskId]: "NOT_STARTED" });
-        updateQueue.queue({ ...updateQueue.current(), taskItemChecklist: {} } as UserData);
+      if (userData.profileData.industryId !== profileData.industryId) {
+        updateQueue.queue({ taskItemChecklist: {} }).queueTaskProgress({ [naicsCodeTaskId]: "NOT_STARTED" });
       }
 
       updateQueue.queueProfileData(profileData);
 
       (async (): Promise<void> => {
-        updateQueue.queue(await postGetAnnualFilings(updateQueue.current()));
-        updateQueue.update().then(async () => {
-          setIsLoading(false);
-          setAlert("SUCCESS");
-          await redirect({ success: true });
-        });
+        updateQueue
+          .queue(await postGetAnnualFilings(updateQueue.current()))
```

Comment:
this instance of updateQueue.current() is different because it relies on updates that may have happened earlier in the function. This will be addressed in the next PR with BusinessFormation where there are similar issues